mypy/util.py

```diff
@@ -158,18 +160,27 @@ def _replace(src: PathType[AnyStr], dest: PathType[AnyStr], timeout: float = 1)
     Increase wait time exponentially until total wait of timeout sec;
     on timeout, give up and reraise the last exception seen.
     """
-    n_iter = max(1, math.ceil(math.log2(timeout / 0.001)))
-    for i in range(n_iter):
-        # Last wait is ~ timeout/2, so that total wait ~ timeout.
-        wait = timeout / 2 ** (n_iter - i - 1)
+    # Find starting value for wait.
+    wait = timeout
+    total_wait: float = 0
+    while True:
+        if wait < 0.001:
+            break
+        wait /= 2
+    while True:
```

Comment:
Let's just keep it simple rather than trying to work against xdist. If it's hard to figure out now I'm sure it will be hard to maintain in the future.

Just in case, my go-to xdist wizard is @gnprice .